lib/TfIdfVectorizer.jsm

```diff
@@ -0,0 +1,74 @@
+
+const REGEXP_SPLITS = new RegExp(`[${UNICODE_SPACE}${UNICODE_SYMBOL}${UNICODE_PUNCT}]+`);
+const REGEXP_ALPHANUMS = new RegExp(`^[${UNICODE_NUMBER}${UNICODE_MARK}${UNICODE_LETTER}]+$`);
+
+this.TfIdfVectorizer = class TfIdfVectorizer {
+  /**
+   * Downcases the text, and splits it into consecutive alphanumeric characters.
+   * This is locale aware, and so will not strip accents. This uses "word
+   * breaks", and os is not appropriate for languages without them
+   * (e.g. Chinese).
+   */
+  tokenize(text) {
+    return text.toLocaleLowerCase().split(REGEXP_SPLITS).filter(tok => tok.match(REGEXP_ALPHANUMS));
+  }
+
+  /**
+   * Converts a sequence of tokens into an L2 normed TF-IDF. Any terms that are
+   * not preindexed (i.e. does has a computed inverse document frequency) will
+   * be dropped.
+   */
+  toksTotfIdfVector(tokens, vocab_idfs) {
+    let tfidfs = {};
```

Comment:
Again, a `Map` would be better than object for this.

Reply:
What's the difference?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#Objects_and_maps_compared